index.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const Base = require('bfx-facs-base')
+
+class GrcSlack extends Base {
+  constructor (caller, opts, ctx) {
+    super(caller, opts, ctx)
+
+    this.name = 'grc-slack'
+    this._hasConf = true
+    this.init()
+
+    if (opts.conf) this.conf = opts.conf
+  }
+
+  message (message, reqChannel) {
+    if (!this.conf.enable) return false
+    const slack = this.conf
+    const worker = slack.worker || 'rest:ext:slack'
+    const maxLength = slack.max_length || 1024
+    const env = (slack.env) ? `Env: ${slack.env}, ` : ''
+    const rawText = env + message
+    const text = (maxLength) ? rawText.substr(0, maxLength) : rawText
+    const channel = reqChannel || slack.channel
+    const send = [{ channel, text }]
+
+    return this.caller.grc_bfx.req(
```

Comment:
this can be improved further to return a boolean value through promise and callback since it's an async call

Example:
```js

  message (message, reqChannel = null, cb = null) {
    const isCb = typeof cb === 'function'
    if (!this.conf.enable) {
      if (isCb) cb(null, false)
      return Promise.resolve(false)
    }

    const slack = this.conf
    const worker = slack.worker || 'rest:ext:slack'
    const maxLength = slack.max_length || 1024
    const env = (slack.env) ? `Env: ${slack.env}, ` : ''
    const rawText = env + message
    const text = (maxLength) ? rawText.substr(0, maxLength) : rawText
    const channel = reqChannel || slack.channel
    const send = [{ channel, text }]

    return new Promise((resolve) => {
      this.caller.grc_bfx.req(
        worker,
        'postSlackMsg',
        send,
        { timeout: 10000 },
        (err, res) => {
          const isSuccess = err || !(res || {}).ok ? false : true
          if (isCb) cb(null, isSuccess)
          resolve(isSuccess)
        }
      )
    })
  }
```